.github/workflows/ci.yml

```diff
@@ -81,6 +81,9 @@ jobs:
       run: git submodule update --init --depth 1 fuzz/corpora
     - name: localegen
       run: sudo locale-gen tr_TR.UTF-8
+    - name: fipsvendor
+      # Make one fips build use a customized FIPS vendor
+      run: echo "FIPSVENDOR=CI" >> VERSION.dat
```

Comment:
Just to ask, is there a reason that this should be deposited in a version file, rather than just adding it on the Configure command line?  i.e.
```
FIPSVENDOR=CI ./Configure <options>
```
Not opposed to doing the above way, mind, it just seems different from how we do other build environment settings

Reply:
@nhorman FIPS submissions need to customize full buildinfo: vendor, version, build metadata. As in distinct builds/submissions based on the same 3.4.0 code from the same vendor must report themselves with unique version numbers i.e. 3.4.0-r0 & 3.4.0-r1 (apk) 3.4.0-0ubuntu3+Fips1 & 3.4.0-0ubuntu3+Fips2 (deb) and so on. Reading `Configure`:

```perl
collect_information(
    collect_from_file(catfile($srcdir,'VERSION.dat')),
    qr/\s*(\w+)\s*=\s*(.*?)\s*$/ =>
        sub {
            # Only define it if there is a value at all
            if ($2 ne '') {
                my $k = $1;
                my $v = $2;
                # Some values are quoted.  Trim the quotes
                $v = $1 if $v =~ /^"(.*)"$/;
                $version{uc $k} = $v;
            }
        },
    "OTHERWISE" =>
        sub { die "Something wrong with this line:\n$_\nin $srcdir/VERSION.dat" },
    );

$config{major} = $version{MAJOR} // 'unknown';
$config{minor} = $version{MINOR} // 'unknown';
$config{patch} = $version{PATCH} // 'unknown';
$config{prerelease} =
    defined $version{PRE_RELEASE_TAG} ? "-$version{PRE_RELEASE_TAG}" : '';
$config{build_metadata} =
    defined $version{BUILD_METADATA} ? "+$version{BUILD_METADATA}" : '';
$config{shlib_version} = $version{SHLIB_VERSION} // 'unknown';
$config{release_date} = $version{RELEASE_DATE} // 'xx XXX xxxx';

$config{version} = "$config{major}.$config{minor}.$config{patch}";
$config{full_version} = "$config{version}$config{prerelease}$config{build_metadata}";
```

It is my understanding to customize provider version/buildinfo one would need to modify VERSION.dat to encode correct package version metadata. Hence setting the vendor name there seems appropriate. As one needs to put in version changes there.

I agree that tweaking VERSION.dat is odd, because that file is committed into the VCS.

Should I try to add new ways to override `PRE_RELEASE_TAG BUILD_METADATA RELEASE_DATE FIPSVENDOR` without touching VERSION.dat?

Reply:
A more realistic example with a complete package version information in build info:

```console
$ openssl list -providers -verbose
Providers:
  base
    name: OpenSSL Base Provider
    version: 3.4.0
    status: active
    build info: 3.4.0
    gettable provider parameters:
      name: pointer to a UTF8 encoded string (arbitrary size)
      version: pointer to a UTF8 encoded string (arbitrary size)
      buildinfo: pointer to a UTF8 encoded string (arbitrary size)
      status: integer (arbitrary size)
  fips
    name: Chainguard FIPS Provider for OpenSSL
    version: 3.4.0
    status: active
    build info: 3.4.0-r4
    gettable provider parameters:
      name: pointer to a UTF8 encoded string (arbitrary size)
      version: pointer to a UTF8 encoded string (arbitrary size)
      buildinfo: pointer to a UTF8 encoded string (arbitrary size)
      status: integer (arbitrary size)
      security-checks: integer (arbitrary size)
      tls1-prf-ems-check: integer (arbitrary size)
      drbg-no-trunc-md: integer (arbitrary size)
```

Reply:
No, I wasn't looking for a specific change, just an explination.  Thank you!

Reply:
@nhorman should I add a bit more documentation in any of the FIPS docs explaining "how to set vendor and build info?"

Reply:
Yeah, I think some commentary in README-FIPS.md would be appropriate.  Thanks you!